ngraph/core/src/function.cpp

```diff
@@ -453,4 +602,29 @@ void Function::remove_parameter(const std::shared_ptr<op::Parameter>& param)
         m_parameters.end());
 }
 
+void Function::add_variables(const VariableVector& variables)
+{
+    m_variables.insert(m_variables.end(), variables.begin(), variables.end());
+}
+
+void Function::remove_variable(const VariablePtr& variable)
+{
+    m_variables.erase(std::remove_if(m_variables.begin(),
+                                     m_variables.end(),
+                                     [&variable](VariablePtr& v) { return v == variable; }),
+                      m_variables.end());
+}
+
+VariablePtr Function::get_variable_by_id(const string& variable_id) const
+{
+    auto variable = std::find_if(
+        m_variables.begin(), m_variables.end(), [&variable_id](const VariablePtr& cur) {
+            return cur->get_info().variable_id == variable_id;
+        });
+    if (variable != m_variables.end())
+        return *variable;
+    else
+        return VariablePtr();
```

Comment:
Should we return nullptr here?

Reply:
it's already nullptr
![image](https://user-images.githubusercontent.com/21181928/118432803-3a2fb000-b6e2-11eb-90db-92d5fc85788b.png)